test-utils/runtime/network/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+use std::sync::Arc;
+use futures::prelude::*;
+use sc_network::{config, Event, NetworkService, NetworkWorker};
+use sp_runtime::traits::{Block as BlockT, Header as _};
+use substrate_test_runtime_client::{TestClientBuilder, TestClientBuilderExt as _};
+
+pub type TestNetworkService = NetworkService<
+	substrate_test_runtime_client::runtime::Block,
+	substrate_test_runtime_client::runtime::Hash,
+>;
+
+/// Builds a full node to be used for testing. Returns the node service and its associated events
+/// stream.
+///
+/// > **Note**: We return the events stream in order to not possibly lose events between the
+/// >			construction of the service and the moment the events stream is grabbed.
+pub fn build_test_full_node(config: config::NetworkConfiguration)
```

Comment:
This file contains duplicate codes which also lives in `client/network/src/gossip/tests.rs` and `client/network/src/service/tests.rs`. I tried to reuse this function but failed with error:
```
expected struct `service::NetworkService`, found struct `sc_network::service::NetworkService`
expected enum `sc_network::protocol::event::Event`, found enum `protocol::event::Event`
```
Any help appreciate.

Reply:
This should not block this PR if I understand it correctly.

Also posted [here](https://users.rust-lang.org/t/cyclic-dev-dependency-get-type-mismatch-error/48391) to get more help.

Reply:
Duplicate code isn't bad by itself. Unless there is a reason to remove the duplicate code, I'd rather just keep it as it is.